### DISCUSSION TITLE
Clean and Efficient Error Handling in NextJS

### DISCUSSION THREAD

**Discussion by developedbynick** [vercel/next.js#65250] [App Router]
Hello There. I'm new to NextJS and coming from the MERN stack. 

In express there's a global error handler, but there's also a package called express-async-handler that catches any errors in route handler and calls a global error handler with the necessary data. 

With all this said, Is there any way to handle errors in an API route cleanly rather than doing try/catch?

**Comment by solo-samurai** [comment]
hello @developedbynick, I haven't used it lately but if you like writing your API in the style of express
check out https://github.com/hoangvvo/next-connect

they support app router pages router and you can handle errors just like express

**Reply by developedbynick** [reply]
Thanks for your reply. I'll wait to see if I get anymore response.
